src/python/module/z5py/dataset.py

```diff
@@ -234,38 +272,48 @@ def compression_options(self):
     def __len__(self):
         return self._impl.len
 
-    # TODO support ellipsis
     def index_to_roi(self, index):
+        try:
+            index_lst = list(index)
+        except TypeError:
+            index_lst = [index]
 
-        # check index types of index and normalize the index
-        if not isinstance(index, (slice, tuple)):
-            raise TypeError("Index must be slice or tuple of slices")
-        index_ = (index,) if isinstance(index, slice) else index
-
-        # check lengths of index
-        if len(index_) > self.ndim:
+        if len([item for item in index_lst if item != Ellipsis]) > self.ndim:
             raise TypeError("Argument sequence too long")
-
-        # check the individual slices
-        if not all(isinstance(ii, slice) for ii in index_):
-            raise TypeError("Index must be slice or tuple of slices")
-        if not all(ii.step is None for ii in index_):
-            raise TypeError("Slice with non-trivial step is not supported")
-        # get the roi begin and shape from the slicing
-        roi_begin = [
-            (0 if index_[d].start is None else index_[d].start)
-            if d < len(index_) else 0 for d in range(self.ndim)
-        ]
-        roi_shape = tuple(
-            ((self.shape[d] if index_[d].stop is None else index_[d].stop)
-             if d < len(index_) else self.shape[d]) - roi_begin[d] for d in range(self.ndim)
-        )
+        elif len(index_lst) < self.ndim and Ellipsis not in index_lst:
+            index_lst.append(Ellipsis)
```

Comment:
It's also worth noting that in numpy/ hdf5, if all indices are integers, the output will be just the value rather than an array of shape (1,)*ndim. That can either be documented or I can put in a special case to make it behave properly.

Reply:
Adding something like
```python
if out.size == 1:
    return out[(0,) * out.ndim]
```
to `__getitem__` should do the trick.

Reply:
I suppose someone might ask for a slice which ends up having a size of 1 - but something like

```python
try:
    if len(index) == self.ndim and all(isinstance(i, int) for i in index):
        return out[(0,) * out.ndim]
except TypeError:
    pass
```

would be fine.